label number:  5
shareLocationOnce 0
sendAudio 1
sendText 2
transferFile 3
sendImage 4

Begin to generate features.
  0%|          | 0/5 [00:00<?, ?it/s]generate features:   0%|          | 0/5 [00:00<?, ?it/s]
  0%|          | 0/5 [00:00<?, ?it/s][A  0%|          | 0/5 [00:00<?, ?it/s]
len of target_all_files of ../../traffic_split/qq/shareLocationOnce: 3155
len of r_files: 100
Traceback (most recent call last):
  File "/home/changc/miniconda3/envs/trafficformer/lib/python3.12/site-packages/flowcontainer/reader.py", line 96, in read
    process = Popen(command, stdout=PIPE, stderr=PIPE)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/changc/miniconda3/envs/trafficformer/lib/python3.12/subprocess.py", line 1028, in __init__
    self._execute_child(args, executable, preexec_fn, close_fds,
  File "/home/changc/miniconda3/envs/trafficformer/lib/python3.12/subprocess.py", line 1963, in _execute_child
    raise child_exception_type(errno_num, err_msg, err_filename)
FileNotFoundError: [Errno 2] No such file or directory: 'tshark'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/changc/miniconda3/envs/trafficformer/lib/python3.12/site-packages/flowcontainer/reader.py", line 100, in read
    raise  EnvironmentError('tshark is not installed or added to environment path.')
OSError: tshark is not installed or added to environment path.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/changc/TrafficFormer/data_generation/finetuning_data_gen.py", line 374, in process_one_label
    feature_data = get_feature_flow(r_f, select_packet_len=payload_length, packets_num=payload_packet,start_index=start_index)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/changc/TrafficFormer/data_generation/finetuning_data_gen.py", line 175, in get_feature_flow
    feature_result = extract(label_pcap, filter='tcp', extension=['tls.record.content_type', 'tls.record.opaque_type', 'tls.handshake.type'])
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/changc/miniconda3/envs/trafficformer/lib/python3.12/site-packages/flowcontainer/extractor.py", line 75, in extract
    result = real_extract(infile,filter,extension, ip_layer, cmd_parameter, verbose)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/changc/miniconda3/envs/trafficformer/lib/python3.12/site-packages/flowcontainer/extractor.py", line 16, in real_extract
    result = reader.read(infile,filter,extension,ip_layer,cmd_parameter=cmd_parameter)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/changc/miniconda3/envs/trafficformer/lib/python3.12/site-packages/flowcontainer/reader.py", line 115, in read
    raise EnvironmentError(ex)
OSError: tshark is not installed or added to environment path.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/changc/TrafficFormer/data_generation/finetuning_data_gen.py", line 747, in <module>
    fire.Fire(main)
  File "/home/changc/miniconda3/envs/trafficformer/lib/python3.12/site-packages/fire/core.py", line 135, in Fire
    component_trace = _Fire(component, args, parsed_flag_args, context, name)
                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/changc/miniconda3/envs/trafficformer/lib/python3.12/site-packages/fire/core.py", line 468, in _Fire
    component, remaining_args = _CallAndUpdateTrace(
                                ^^^^^^^^^^^^^^^^^^^^
  File "/home/changc/miniconda3/envs/trafficformer/lib/python3.12/site-packages/fire/core.py", line 684, in _CallAndUpdateTrace
    component = fn(*varargs, **kwargs)
                ^^^^^^^^^^^^^^^^^^^^^^
  File "/home/changc/TrafficFormer/data_generation/finetuning_data_gen.py", line 743, in main
    generation_multiP(pcap_split_path, samples_per_class, tmp_path)
  File "/home/changc/TrafficFormer/data_generation/finetuning_data_gen.py", line 444, in generation_multiP
    process_one_label(session_pcap_path,key,payload_length,payload_packet,samples,label_id,start_index)
  File "/home/changc/TrafficFormer/data_generation/finetuning_data_gen.py", line 377, in process_one_label
    exc_type, exc_value, exc_traceback = sys.exc_info()
                                         ^^^
NameError: name 'sys' is not defined. Did you forget to import 'sys'?
generate features:   0%|          | 0/5 [00:00<?, ?it/s]
